src/panels/NightSkyPanel/components/IconLabelButton.tsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from 'react';
+import { Checkbox, Stack, Text } from '@mantine/core';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { useProperty } from '@/hooks/properties';
+import {
+  AbcIcon,
+  BandIcon,
+  CompassLargeIcon,
+  CompassMarksIcon,
+  CompassSmallIcon,
+  HomeIcon,
+  LineIcon,
+  PaintBrushIcon,
+  PencilIcon,
+  SingleDotIcon,
+  SphereIcon
+} from '@/icons/icons';
+
+interface Props {
+  title: string;
+  icon: string;
+  identifier?: string;
+  onAction?: string;
+  offAction?: string;
+  boolProp?: string;
+  directionCheck?: string;
+}
+
+export function IconLabelButton({
```

Comment:
You've gotten some comments below, but this component needs to be simplified, for example using an existing component for the toggle checkbox.